seam/seam.py

```diff
@@ -37,6 +40,10 @@ def __init__(
           The API endpoint to which the request should be sent.
         wait_for_action_attempt : bool or dict, optional
           Controls whether to wait for an action attempt to complete, either as a boolean or as a dictionary specifying `timeout` and `poll_interval`. Defaults to `False`.
+        client : requests.Session, optional
+          A pre-configured requests session to be used for making HTTP requests. If not provided, a new `SeamHttpClient` instance will be created using the `client_options` and other relevant parameters.
+        client_options : dict, optional
+          A dictionary of options that will be passed to the `SeamHttpClient` constructor when initializing a new requests session client. This allows for customization of the HTTP client, such as setting additional headers or configuring timeouts. For detailed information on available options, refer to the niquests library [repo](https://github.com/jawah/niquests). If client is provided, this parameter will be ignored.
```

Comment:
Let's remove the `client` and `client_options` parameters for now. Since niquests has a Session and Request options, I need to think a little more about how we should expose this, I'm still learning about the niquest API.

Once this is merged in and the generated code is updated to use the new client, we can merge the blocked PR. Then it will be a lot easier for me to experiment with this and contribute.

Reply:
47145c5bcfc985007236f77c6dfb3a739c2c09c4